src/qt/managenamespage.cpp

```diff
@@ -0,0 +1,143 @@
+#include <qt/managenamespage.h>
+#include <qt/forms/ui_managenamespage.h>
+
+#include <qt/csvmodelwriter.h>
+#include <qt/guiutil.h>
+#include <qt/nametablemodel.h>
+#include <qt/platformstyle.h>
+#include <qt/walletmodel.h>
+
+#include <QMessageBox>
+#include <QMenu>
+#include <QSortFilterProxyModel>
+
+ManageNamesPage::ManageNamesPage(const PlatformStyle *platformStyle, QWidget *parent) :
+    QWidget(parent),
+    platformStyle(platformStyle),
+    ui(new Ui::ManageNamesPage),
+    model(nullptr),
+    walletModel(nullptr),
+    proxyModel(nullptr)
+{
+    ui->setupUi(this);
+
+    // Context menu actions
+    QAction *copyNameAction = new QAction(tr("Copy &Name"), this);
+    QAction *copyValueAction = new QAction(tr("Copy &Value"), this);
+
+    // Build context menu
+    contextMenu = new QMenu();
+    contextMenu->addAction(copyNameAction);
+    contextMenu->addAction(copyValueAction);
+
+    // Connect signals for context menu actions
+    connect(copyNameAction, &QAction::triggered, this, &ManageNamesPage::onCopyNameAction);
+    connect(copyValueAction, &QAction::triggered, this, &ManageNamesPage::onCopyValueAction);
+
+    connect(ui->tableView, &QTableView::customContextMenuRequested, this, &ManageNamesPage::contextualMenu);
+    ui->tableView->setEditTriggers(QAbstractItemView::NoEditTriggers);
+
+    ui->tableView->installEventFilter(this);
+}
+
+ManageNamesPage::~ManageNamesPage()
+{
+    delete ui;
+}
+
+void ManageNamesPage::setModel(WalletModel *walletModel)
+{
+    this->walletModel = walletModel;
+    model = walletModel->getNameTableModel();
+
+    proxyModel = new QSortFilterProxyModel(this);
+    proxyModel->setSourceModel(model);
+    proxyModel->setDynamicSortFilter(true);
+    proxyModel->setSortCaseSensitivity(Qt::CaseInsensitive);
+    proxyModel->setFilterCaseSensitivity(Qt::CaseInsensitive);
+
+    ui->tableView->setModel(proxyModel);
+    ui->tableView->sortByColumn(0, Qt::AscendingOrder);
+
+    ui->tableView->horizontalHeader()->setHighlightSections(false);
+
+    // Set column widths
+    ui->tableView->horizontalHeader()->resizeSection(
+            NameTableModel::Name, 320);
+    ui->tableView->horizontalHeader()->setSectionResizeMode(QHeaderView::Stretch);
+
+    connect(ui->tableView->selectionModel(), &QItemSelectionModel::selectionChanged,
+            this, &ManageNamesPage::selectionChanged);
+
+    selectionChanged();
+}
+
+bool ManageNamesPage::eventFilter(QObject *object, QEvent *event)
+{
+    if (event->type() == QEvent::FocusIn)
+    {
+        if (object == ui->tableView)
+        {
```

Comment:
Should there be something in here?  If not, then you should remove the two `if` as well here (and perhaps add them in later together with a change that actually puts conditional code here).

Reply:
This `if` statement is going to contain stuff for the the Submit Name and Configure Name buttons, which aren't currently present in this PR but will be added later.  I can factor out the `if` into a separate commit and move that commit into a separate PR.

Reply:
Moved this code to https://github.com/namecoin/namecoin-core/pull/376